query-node/mappings/council.ts

```diff
@@ -0,0 +1,928 @@
+*/
+export async function council_CandidacyStakeRelease({ event, store }: EventContext & StoreContext): Promise<void> {
+  // common event processing
+
+  const [memberId] = new Council.CandidacyStakeReleaseEvent(event).params
+  const candidate = await getCandidate(store, memberId.toString()) // get last member's candidacy record
+
+  const candidacyStakeReleaseEvent = new CandidacyStakeReleaseEvent({
+    ...genericEventFields(event),
+    candidate,
+  })
+
+  await store.save<CandidacyStakeReleaseEvent>(candidacyStakeReleaseEvent)
+
+  // specific event processing
+
+  // update candidate info about stake lock
+  candidate.stakeLocked = false
+  await store.save<Candidate>(candidate)
+}
+
+/*
+  The event is emitted when the member is revoking its candidacy during a candidacy announcement stage.
+*/
+export async function council_CandidacyWithdraw({ event, store }: EventContext & StoreContext): Promise<void> {
+  // common event processing
+
+  const [memberId] = new Council.CandidacyWithdrawEvent(event).params
+  const candidate = await getCandidate(store, memberId.toString())
+
+  const candidacyWithdrawEvent = new CandidacyWithdrawEvent({
+    ...genericEventFields(event),
+    candidate,
+  })
+
+  await store.save<CandidacyWithdrawEvent>(candidacyWithdrawEvent)
+
+  // specific event processing
+
+  // mark candidacy as withdrawn
+  const electionRound = await getCurrentElectionRound(store)
+  candidate.candidacyWithdrawn = true
+  await store.save<Candidate>(candidate)
+}
+
+/*
+  The event is emitted when the candidate changes its candidacy note.
+*/
+export async function council_CandidacyNoteSet({ event, store }: EventContext & StoreContext): Promise<void> {
+  // common event processing
+
+  const [memberId, note] = new Council.CandidacyNoteSetEvent(event).params
+
+  // load candidate recored
+  const electionRound = await getCurrentElectionRound(store)
+  const candidate = await getCandidate(store, memberId.toString(), electionRound)
+
+  // unpack note's metadata and save it to db
+  const metadata = deserializeMetadata(CouncilCandidacyNoteMetadata, note)
+  const noteMetadata = candidate.noteMetadata
+  // `XXX || (null as any)` construct clears metadata if requested (see https://github.com/Joystream/hydra/issues/435)
+  noteMetadata.header = isSet(metadata?.header) ? metadata?.header || (null as any) : metadata?.header
+  noteMetadata.bulletPoints = metadata?.bulletPoints || []
+  noteMetadata.bannerImageUri = isSet(metadata?.bannerImageUri)
+    ? metadata?.bannerImageUri || (null as any)
+    : metadata?.bannerImageUri
+  noteMetadata.description = isSet(metadata?.description)
+    ? metadata?.description || (null as any)
+    : metadata?.description
+  await store.save<CandidacyNoteMetadata>(noteMetadata)
+
+  // save metadata set by this event
+  const noteMetadataSnapshot = new CandidacyNoteMetadata({
+    header: metadata?.header || undefined,
```

Comment:
`header === ''` and `header === undefined` will have different meaning (the first one will unset the header, while the second one will not change it), but if you use `header || undefined`, it will result in `undefined` in both these cases.
You should probably use:
```
    header: metadata?.header ?? undefined,
```

Reply:
Fixed.